### PR TITLE
JENA-2039: Fix and tests (IRI handling), and more IRI cleanup (JENA-2031)

### PR DESCRIPTION
**Pull request by afs**

### CODE REVIEW

File: jena-arq/src/main/java/org/apache/jena/riot/checker/CheckerIRI.java [hunk 1]
```diff
@@ -86,87 +93,42 @@ public static void iriViolations(IRI iri, ErrorHandler errorHandler) {
...
                 boolean isError = v.isError() ;
 
-                // Ignore these. They seem to present even if switched off.
-                if ( code == Violation.LOWERCASE_PREFERRED
-                     || code == Violation.PERCENT_ENCODING_SHOULD_BE_UPPERCASE
-                     || code == Violation.SCHEME_PATTERN_MATCH_FAILED )
-                    continue ;
-
                 // Anything we want to reprioritise?
+                // Some jena-iri violations are fully controllable by error.wraning controls
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/checker/CheckerIRI.java:115]: s/wraning/warning, but not important as it's a code comment (not javadoc)

File: jena-arq/src/main/java/org/apache/jena/riot/system/Checker.java [hunk 2]
```diff
@@ -21,44 +21,38 @@
...
-        checkBlankNodes = new CheckerBlankNodes(handler) ;
-        checkVars = new CheckerVar(handler) ;        
-    }
-
-    public ErrorHandler getHandler()                { return handler ; } 
-    public void setHandler(ErrorHandler handler)    { this.handler = handler ; }
-    
-    public boolean check(Node node, long line, long col)
-    {
+            handler = ErrorHandlerFactory.getDefaultErrorHandler();
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/system/Checker.java:45]: Extra spaces?

File: jena-arq/src/main/java/org/apache/jena/riot/system/IRIResolver.java [hunk 3]
```diff
@@ -29,19 +29,63 @@
...
+        setErrorWarning(iriCheckerInst, ViolationCodes.NOT_NFKC, false, false);
+
+        // ** Applies to various unicode blocks.
+        setErrorWarning(iriCheckerInst, ViolationCodes.COMPATIBILITY_CHARACTER, false, true);
+        setErrorWarning(iriCheckerInst, ViolationCodes.UNDEFINED_UNICODE_CHARACTER, false, true);
+        // The set of legal characters depends on the Java version.
+        // If not set, this causes test failures in Turtle and Trig eval tests.
+        setErrorWarning(iriCheckerInst, ViolationCodes.UNASSIGNED_UNICODE_CHARACTER, false, false);
+
+        if ( ShowResolverSetup ) {
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/system/IRIResolver.java:80]: As we are changing it, could/should we also rename it to `showResolverSetup`?

**Review by kinow — APPROVED** [review]

File: jena-arq/src/main/java/org/apache/jena/riot/system/Checker.java (hunk 2 above)
Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/riot/system/Checker.java:45]: The previous line is `if ( handler == null )` -- the diff doesn't make it very clear, it's decided that's an unchanged line as seval changes happen in the same area.

Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/riot/system/Checker.java:45]: Will fix directly along with some clearing up.

**Pull request merged**